controller/core/pattern_selector.py

```diff
@@ -96,13 +127,28 @@ def deactivateButton(self, button_name):
 
     @run_in_executor
     def poll(self):
-        button_event = self.launchpad.panel.buttons().poll_for_event()
-        if button_event and button_event.type == lpminimk3.ButtonEvent.PRESS:
-            self.buttons_pressed.append(button_event.button.name)
-        elif button_event and button_event.type == lpminimk3.ButtonEvent.RELEASE:
-            pass
-
-    async def launchpadListener(self):
+        
+        if self.dmx:
+            #Check for waiting messages from DMX controller
+            if self.dmx.inWaiting() > 0:
```

Comment:
Are you adding these inWaiting blocks here to not block on read_until? It should be fine to block if we put the DMX listener into it's own coroutine that is added in main.py

Reply:
I was unsure about that -- I thought we might want to aggregate all controller input and apply once per loop. Will revise

Reply:
self.launchpad.panel.buttons().poll_for_event() below is also blocking until a button is pressed. So we have to break this up anyway if we want to have both controllers connected.